### PR TITLE
Feature/clean tabpanel signals and report

### PR DESCRIPTION
**Pull request by tinneuro**
During a debugging process of the reports I realised that it is difficult for me to generate all params needed to run the rmd report. Thus I decided to create a `pad_signals()` function to be able to generate data needed for the report also without using the app and improve the readability of our signals server side.
In this process I realised that some namings of me are not intuitive and I had struggles to understand some objects and input parameters needed for the report. Thus this PR does some cleaning as well

- adding of a function called `pad_signals()` and removing the code from the signals tab doing this and replacing it by this function
- adding of method and number_of_weeks to the results of the get_signals() function. This reduces the number of parameters which need to be carried and forwarded around in our app. Thus removed some function parameters in the report server. Furthermore if someone else wants to use the get_signals() outcome this person also has the informaton of the algorithm attached when running for example over several algorithms
- renaming of data to filtered_data in multiple instances

Question: 
I have struggles to understand what exactly signal_data() which is generated inside the mod_tabpanel_signals.R and is used for the report is doing. Can somebody help me to clarify what this does and for what this is needed? Maybe we can then improve on adding comments and/or changing its name.

### CODE REVIEW

**Comment by alchalu** [comment]
I am not totally sure if this answers your question, but the original idea of the `data` input for the report function was that you could just feed the original data input dataframe to the `run_report` function and all the calculation steps would be done in the report (only if no signal_results and signals_agg dataframes were not available). So that the `run_report()` function could be used as a standalone function. However all the dataprocessing steps were scattered throught the app, so I did not really implement that option. Now that you put everything in the `pad_signals` function it's a possibility. Thank you! 

Maybe we do not need the `signal_data()` at all then? I am pretty sure we do not need it for the report.

**Comment by tinneuro** [comment]
Ok makes sense thanks for this enlightenment. I was wondering because there is some filtering by dates involved in the process of generating the `signal_data() ` thus I thought it is not the same as if I just forward the preprocessed input dataset. But I can try it out if it still works correctly.

**Comment by alchalu** [comment]
Looking at it a bit closer I think `signal_data` is something residual from a previous version of the report and it would probably best to just delete it from the app. And hopefully with the help of `pad_signals()` the `run_report` could be used as a standalone function.

**Comment by tinneuro** [comment]
Ok great I will try it out tomorrow and add this change to this PR.

File: inst/report/SignalDetectionReport.Rmd
Comment (alchalu) [inline inst/report/SignalDetectionReport.Rmd:1]: why did you delete the params `number_of_weeks` and `algo`?

File: inst/report/SignalDetectionReport_body.Rmd
Comment (alchalu) [inline inst/report/SignalDetectionReport_body.Rmd:1]: ah I see, I think I am still confused how we want the run_report function to be used. Do we want the user to have to always create signals_padded and signals_agg? Then we can remove the param `data` as well and also take the data input out of the `run_report` function

**Comment by alchalu** [comment]
Other than my two comments this looks great and the new function is super helpful! Thank you so much 🎉

File: inst/report/SignalDetectionReport_body.Rmd
Comment (tinneuro) [reply inst/report/SignalDetectionReport_body.Rmd:1]: Ah yes you are right. I did not think about this. For the report I agree that I will add them back as optional parameters
